src/api/module/history/service/index.ts

```diff
@@ -0,0 +1,26 @@
+import { Payload, History, HistoryType } from '@/api/module/history/entity';
+import { FindAll, Insert, DeleteRecord } from '@/api/module/history/repository/indexeddb';
+import { DateTime } from 'luxon';
+
+export function Index(): Promise<unknown> {
+  return FindAll();
+}
+
+export function Create(payload: Payload): Promise<unknown> {
+  if (!payload.juz || !payload.approachId)
+    return Promise.reject(new Error('Error 422 Unprocessable Entity'));
+
+  const history: History = {
+    historyType: HistoryType.Juz,
+    juz: payload.juz,
+    approachId: payload.approachId,
+    repeat: 1, // Hardcoded to 1 for juz
+    occuredAt: DateTime.now().toJSDate(),
+  };
+
+  return Insert(history);
+}
+
+export function Destroy(item: History): Promise<unknown> {
```

Comment:
need to rename to DestroyHistory etc for easier tracking. can be renamed here or in the consuming component

Reply:
this same like comment above too